OCR/ocr/services/image_segmenter.py

```diff
@@ -79,33 +76,42 @@ def segment_by_color_bounding_box(self) -> dict[str, np.ndarray]:
 class ImageSegmenter:
     def __init__(
         self,
-        raw_image,
-        segmentation_template,
-        labels,
         segmentation_function=segment_by_mask_then_crop,
         debug=False,
     ):
-        self.debug = debug
         self.segmentation_function = segmentation_function
+        self.debug = debug
 
-        if not os.path.isfile(raw_image) or not os.path.isfile(segmentation_template):
+    def segment(
+        self,
+        raw_image,
+        segmentation_template,
+        labels,
+    ) -> dict[str, np.ndarray]:
+        return self.segmentation_function(raw_image, segmentation_template, labels, self.debug)
+
+    def load_and_segment(self, raw_image_path, segmentation_template_path, labels_path):
+        if (
+            not os.path.isfile(raw_image_path)
+            or not os.path.isfile(segmentation_template_path)
+            or not os.path.isfile(labels_path)
+        ):
             raise FileNotFoundError("One or more input files do not exist.")
 
-        self.raw_image = cv.imread(raw_image)
-        if self.raw_image is None:
-            raise ValueError(f"Failed to open image file: {raw_image}")
+        raw_image = cv.imread(raw_image_path)
+        if raw_image is None:
+            raise ValueError(f"Failed to open image file: {raw_image_path}")
 
-        self.segmentation_template = cv.imread(segmentation_template)
-        if self.segmentation_template is None:
-            raise ValueError(f"Failed to open image file: {segmentation_template}")
+        segmentation_template = cv.imread(segmentation_template_path)
+        if segmentation_template is None:
+            raise ValueError(f"Failed to open image file: {segmentation_template_path}")
 
-        self.labels = labels
+        labels = json.load(open(labels_path, "r"))
 
         if self.debug is True:
             self.debug_folder = "debug_segments"
             os.makedirs(self.debug_folder, exist_ok=True)
-            print(f"raw_image shape: {self.raw_image.shape}")
-            print(f"segmentation_template shape: {self.segmentation_template.shape}")
+            print(f"raw_image shape: {raw_image.shape}")
```

Comment:
In a following tickets we should change these to logging
logging.debug(f"raw_image shape: {self.raw_image.shape}")